### PR TITLE
PoS NetworkManager Address

### PR DESCRIPTION
**Pull request by AeonSw4n**

### CODE REVIEW

**Comment by AeonSw4n** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/957?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `feature/pos-networking-and-syncing`
  * **PR #942** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/942?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #943** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/943?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #944** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/944?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #945** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/945?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #957** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/957?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
          * **PR #912** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/912?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/957?utm_source=stack-comment).

File: lib/server.go [hunk 1]
```diff
@@ -2488,27 +2511,36 @@ func (srv *Server) _startAddressRelayer() {
...
+		glog.V(2).Infof("Server.startAddressRelayer: Found %d addrs to "+
 			"relay: %v", len(addrsToBroadcast), spew.Sdump(addrsToBroadcast))
 		// Iterate over all our peers and broadcast the addrs to all of them.
-		for _, pp := range srv.cmgr.GetAllPeers() {
-			pp.AddDeSoMessage(&MsgDeSoAddr{
+		for _, rn := range remoteNodes {
+			if !rn.IsHandshakeCompleted() {
+				continue
+			}
+			if err := rn.SendMessage(&MsgDeSoAddr{
```
Comment (tholonious) [inline lib/server.go:2538]: nit, clean up nesting for readability
```suggestion
			addrMsg := MsgDeSoAddr{
```

File: lib/server.go [hunk 2]
```diff
@@ -2488,27 +2511,36 @@ func (srv *Server) _startAddressRelayer() {
...
 				AddrList: addrsToBroadcast,
-			}, false)
+			}); err != nil {
```
Comment (tholonious) [inline lib/server.go:2540]:
```suggestion
			if err := rn.SendMessage(&addrMsg); err != nil {
```

File: lib/server.go [hunk 3]
```diff
@@ -2468,18 +2485,24 @@ func (srv *Server) _startAddressRelayer() {
...
+					continue
+				}
+				bestAddress := srv.AddrMgr.GetBestLocalAddress(netAddr)
 				if bestAddress != nil {
-					glog.V(2).Infof("Server.Start._startAddressRelayer: Relaying address %v to "+
-						"peer %v", bestAddress.IP.String(), pp)
-					pp.AddDeSoMessage(&MsgDeSoAddr{
+					glog.V(2).Infof("Server.startAddressRelayer: Relaying address %v to "+
+						"RemoteNode (id= %v)", bestAddress.IP.String(), rn.GetId())
+					if err := rn.SendMessage(&MsgDeSoAddr{
```
Comment (tholonious) [inline lib/server.go:2505]: nit: clean up nesting for readability
```suggestion
					addrMsg := MsgDeSoAddr{
```

File: lib/server.go [hunk 4]
```diff
@@ -2488,27 +2511,36 @@ func (srv *Server) _startAddressRelayer() {
 								Services:  (ServiceFlag)(bestAddress.Services),
 							},
 						},
-					}, false)
+					}); err != nil {
```
Comment (tholonious) [inline lib/server.go:2514]:
```suggestion
					if err := rn.SendMessage(&addrMsg}); err != nil {
```

File: lib/server.go [hunk 5]
```diff
@@ -2161,20 +2166,33 @@ func (srv *Server) StartStatsdReporter() {
 	}()
 }
 
-func (srv *Server) _handleAddrMessage(pp *Peer, msg *MsgDeSoAddr) {
+func (srv *Server) _handleAddrMessage(origin *Peer, desoMsg DeSoMessage) {
```
Comment (tholonious) [inline lib/server.go:2169]: nit: pp is the common and accepted variable name for the Peer type in incoming messages handlers. Revert this for consistency with the rest of the code
```suggestion
func (srv *Server) _handleAddrMessage(pp *Peer, desoMsg DeSoMessage) {
```

**Review by tholonious — CHANGES_REQUESTED** [review]

File: lib/connection_manager.go [hunk 6]
```diff
@@ -105,9 +105,8 @@ type ConnectionManager struct {
 	// peers' time.
 	timeSource chainlib.MedianTimeSource
 
-	// Events that can happen to a peer.
-	newPeerChan  chan *Peer
```
[the remark this answers is not in the document]
Comment (AeonSw4n) [reply lib/connection_manager.go:109]: It's redundant now with the NetworkManager design. Peer is created directly from an established outbound/inbound connection in the same thread, and there is no need for the `newPeerChan` indirection.

**Pull request merged**